sshd-core/src/test/java/org/apache/sshd/common/config/keys/OpenSSHKeyUtilsTest.java

```diff
@@ -0,0 +1,130 @@
+            this.hashes = hashes;
+        }
+    }
+
+    private static KeyEntry[] KEY_ENTRIES = new KeyEntry[] {
+        new KeyEntry(
+            "ssh-rsa AAAAB3NzaC1yc2EAAAABIwAAAQEAxr3N5fkt966xJINl0hH7Q6lLDRR1D0yMjcXCE5roE9VFut2ctGFuo90TCOxkPOMnwzwConeyScVF4ConZeWsxbG9VtRh61IeZ6R5P5ZTvE9xPdZBgIEWvU1bRfrrOfSMihqF98pODspE6NoTtND2eglwSGwxcYFmpdTAmu+8qgxgGxlEaaCjqwdiNPZhygrH81Mv2ruolNeZkn4Bj+wFFmZTD/waN1pQaMf+SO1+kEYIYFNl5+8JRGuUcr8MhHHJB+gwqMTF2BSBVITJzZUiQR0TMtkK6Vbs7yt1F9hhzDzAFDwhV+rsfNQaOHpl3zP07qH+/99A0XG1CVcEdHqVMw== lgoldstein@LGOLDSTEIN-WIN7",
+            new Pair<>(
+                (DigestFactory)BuiltinDigests.md5,
+                "MD5:24:32:3c:80:01:b3:e1:fa:7c:53:ca:e3:e8:4e:c6:8e"
+            ),
+            new Pair<>(
+                (DigestFactory)BuiltinDigests.sha256,
+                "SHA256:1wNOZO+/XgNGJMx8UUJst33V+bBMTz5EcL0B6y2iRv0"
+            )
+        ),
+        new KeyEntry(
+            "ssh-dss AAAAB3NzaC1kc3MAAACBAMg/IxsG5BxnF5gM7IKqqR0rftxZC+n5GlbO+J4H+iIb/KR8NBehkxG3CrBZMF96M2K1sEGYLob+3k4r71oWaPul8n5rt9kpd+JSq4iD2ygOyg6Kd1/YDBHoxneizy6I/bGsLwhAAKWcRNrXmYVKGzhrhvZWN12AJDq2mGdj3szLAAAAFQD7a2MltdUSF7FU3//SpW4WGjZbeQAAAIBf0nNsfKQL/TEMo7IpTrEMg5V0RnSigCX0+yUERS42GW/ZeCZBJw7oL2XZbuBtu63vMjDgVpnb92BdrcPgjJ7EFW6DlcyeuywStmg1ygXmDR2AQCxv0eX2CQgrdUczmRa155SDVUTvTQlO1IyKx0vwKAh1H7E3yJUfkTAJstbGYQAAAIEAtv+cdRfNevYFkp55jVqazc8zRLvfb64jzgc5oSJVc64kFs4yx+abYpGX9WxNxDlG6g2WiY8voDBB0YnUJsn0kVRjBKX9OceROxrfT4K4dVbQZsdt+SLaXWL4lGJFrFZL3LZqvySvq6xfhJfakQDDivW4hUOhFPXPHrE5/Ia3T7A= dsa-key-20130709",
+            new Pair<>(
+                (DigestFactory)BuiltinDigests.md5,
+                "MD5:fb:29:14:8d:94:f9:1d:cf:6b:0e:a4:35:1d:83:44:2f"
+            ),
+            new Pair<>(
+                (DigestFactory)BuiltinDigests.sha256,
+                "SHA256:grxw4KhY1cK6eOczBWs7tDVvo9V0PQw4E1wN1gJvHlw"
+            )
+        ),
+        new KeyEntry(
+            "ecdsa-sha2-nistp384 AAAAE2VjZHNhLXNoYTItbmlzdHAzODQAAAAIbmlzdHAzODQAAABhBFImZtcTj842stlcVHLFBFxTEx7lu3jW9aZCvd0r9fUNKZ6LbRPh6l1oJ4ozArnw7XreQBUc5oNd9HB5RNJ8jl1nWXY5cXBA7McZrKZrYmk+zxNhH6UL+kMLaJkyngJHQw== root@osv-linux",
+            new Pair<>(
+                (DigestFactory)BuiltinDigests.md5,
+                "MD5:e6:dc:a2:4f:5b:11:b2:3c:0f:e8:f6:d8:d1:01:e9:d3"
+            ),
+            new Pair<>(
+                (DigestFactory)BuiltinDigests.sha512,
+                "SHA512:4w6ZB78tmFWhpN2J50Ok6WeMJhZp1X0xN0EKWxZmRLcYDbCWhyJDe8lgrQKWqdTCMZ5aNEBl9xQUklcC5Gt2jg"
+            )
+        )
+    };
+
+    public OpenSSHKeyUtilsTest() {
+        super();
+    }
+
+    @BeforeClass
+    public static void beforeClass() throws IOException, GeneralSecurityException {
+        for (KeyEntry entry : KEY_ENTRIES) {
+            try {
+                entry.key = PublicKeyEntry.parsePublicKeyEntry(entry.keyString).resolvePublicKey();
+            } catch (NoSuchProviderException e) {
+                // ignore key
+            }
+        }
```

Comment:
Add an output message indicating this so we'll know it happened when we examine the tests output
```java
} catch (NoSuchProviderException e) { // ignored
       System.out.println("Skip unsupported key: " + entry.keyString);
}
```